src/common/dao/oidc_user_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dao
+
+import (
+	"testing"
+
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOIDCUserMetaDaoMethods(t *testing.T) {
+
+	user111 := &models.User{
+		Username: "user111",
+		Email:    "user111@email.com",
+	}
+	user222 := &models.User{
+		Username: "user222",
+		Email:    "user222@email.com",
+	}
+	err := OnBoardUser(user111)
+	require.Nil(t, err)
+	err = OnBoardUser(user222)
```

Comment:
Please make sure you test the case to onboard different user with same sub, and check the second one is not inserted into `harbor_user` table.